### DISCUSSION TITLE
Is there a way to mark videos in history as watched?

### DISCUSSION THREAD

**Discussion by novuschroma** [FreeTubeApp/FreeTube#4368] [Q&A]
Hi, I'm new to FreeTube and really enjoying it so far except for one thing: I successfully imported my watch history from YouTube, but none of my history was flagged as watched. 

Having the videos isolated on a history page isn't helpful to me. I only care about being able to see whether or not I already watched videos when I'm browsing. Even the menu item to mark videos as watched is missing from the history page, so I can't easily do this manually either.

Is there an option to do this that I'm missing?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @novuschroma 

Could u identify videos in your History page that are also listed on the Subscription page but not flagged as watched?
If u could not see them on the Subscription page we can try to check the following:

Go to Settings Page -> Subscription Settings -> Check if `Hide Videos on Watched` is enabled
If enabled please disable it and go to your Subscription page to check if you see a watched banner on the videos that are in your history

**Reply by novuschroma** [reply]
Thanks for the response. It was helpful knowing this wasn't intended behavior, so I tried importing my history again. It works now. 

The issue was that I have two YouTube accounts that I use for different purposes. I tried importing both of them, not realizing the history overwrites instead of appends. Part of my confusion came from another extension I use to track watch history that always appends instead of overwrites. I used that extension so I could see my watched history from any account.

It would be nice if I could also merge the histories in FreeTube since profiles replace the need for managing separate accounts (and is one of the big advantages of using it). For now it looks like I'll have to choose which history to keep.

**Reply by novuschroma** [reply]
It seems that I spoke too soon. The reason I thought it was overwritten was because after loading the first import, I checked one of my subscriptions and there were 8 videos marked. After the second import, I reloaded the page and all but one of those watch flags were cleared. I figured maybe I'd just accidentally played the one that was still flagged on the wrong account.

However, after looking through some of my other subscriptions, I'm still seeing videos flagged as watched that would be from the first import.

So it seems that some were unflagged, but not all of them. I'm not sure what the pattern is.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Hi @novuschroma apologies for taking so long to get back to you. I want to make it clear that if you import multiple history files, FT will always add everything to your History tab. No videos will be removed from the first import when you imported your second file.
